src/main/java/com/esri/samples/raster/raster_rendering_rule/RasterRenderingRuleSample.java

```diff
@@ -0,0 +1,166 @@
+            stage.show();
+
+            // create drop down menu of Rendering Rules
+            renderingRuleDropDownMenu = new ComboBox<>();
+            renderingRuleDropDownMenu.setPromptText("Select a Raster Rendering Rule");
+            renderingRuleDropDownMenu.setEditable(false);
+            renderingRuleDropDownMenu.setMaxWidth(260.0);
+
+            // create a Streets BaseMap
+            map = new ArcGISMap(Basemap.createStreets());
+
+            // add the map to a new map view
+            mapView = new MapView();
+            mapView.setMap(map);
+
+            // create an Image Service Raster as a raster layer and add to map
+            final ImageServiceRaster imageServiceRaster = new ImageServiceRaster(imageServiceUrl);
+            final RasterLayer imageRasterLayer = new RasterLayer(imageServiceRaster);
+            map.getOperationalLayers().add(imageRasterLayer);
+
+            imageServiceRaster.loadAsync();
+
+            // add event listener to loading of Image Service Raster and wait until loaded
+            imageRasterLayer.addDoneLoadingListener(() -> {
+                if (imageRasterLayer.getLoadStatus() == LoadStatus.LOADED){
+                    // zoom to extent of the raster
+                    mapView.setViewpointGeometryAsync(imageServiceRaster.getServiceInfo().getFullExtent());
+
+                    // get the predefined rendering rules
+                    List<RenderingRuleInfo> renderingRuleInfos = imageServiceRaster.getServiceInfo().getRenderingRuleInfos();
+
+                    //populate the drop down menu with the rendering rule names
+                    populateDropDownMenu(renderingRuleInfos);
+
+                    // listen to selection in the drop-down menu
+                    renderingRuleDropDownMenu.getSelectionModel().selectedItemProperty().addListener(o -> {
+                        // get the index of the selected rendering rule
+                        int selectedRenderingRuleIndex = renderingRuleDropDownMenu.getSelectionModel().getSelectedIndex();
+                        // apply the selected rendering rule
+                        applyRenderingRule(renderingRuleInfos, selectedRenderingRuleIndex);
+                    });
+                }
+            });
+
+            // add the map view to the stack pane
+            stackPane.getChildren().addAll(mapView, renderingRuleDropDownMenu);
+            StackPane.setAlignment(renderingRuleDropDownMenu, Pos.TOP_LEFT);
+            StackPane.setMargin(renderingRuleDropDownMenu, new Insets(10, 0, 0, 10));
+        } catch (Exception e) {
+            // on any error, display the stack trace.
+            e.printStackTrace();
+        }
+    }
+
+    /*
+     * Populate the drop down menu with the rendering rule names
+     * @param renderingRuleInfos    list of rendering rule infos to add to the drop-down
+     */
+
+    private void populateDropDownMenu(List<RenderingRuleInfo> renderingRuleInfos){
```

Comment:
This can be done in the start method. Android samples like to add unnecessary single use methods everywhere for some reason.